internal/zinc-classfile/src/main/scala/sbt/internal/inc/classfile/Analyze.scala

```diff
@@ -55,9 +52,9 @@ private[sbt] object Analyze {
         .orElse(Option(loadedClass.getEnclosingClass).flatMap(binaryToSourceName))
 
       srcClassName match {
-        case Some(srcClassName) =>
-          analysis.generatedNonLocalClass(source, newClass, binaryClassName, srcClassName)
-          classNames += srcClassName
+        case Some(className) =>
+          analysis.generatedNonLocalClass(source, newClass, binaryClassName, className)
+          classNames += className
```

Comment:
Ah right, yeah I have that inspection turned off. This is its motivation:

> Detects a Variable Pattern that shadows a stable identifier defined in the enclosing scope. To perform an equality test against that value, use backticks.
>
> Before:
> ```scala
> val foo = 0
> 0 match {
>     case foo =>
> }
> ```
>
> After:
> ```scala
> val foo = 0
> 0 match {
>     case `foo` =>
> }
> ```

Food for thought :-)